pkg/controllers/statusbackreporter/controller.go

```diff
@@ -0,0 +1,309 @@
+			return
+		}
+
+		// Skip the resource if it is enveloped.
+		idStr := formatWorkResourceIdentifier(&resIdentifier)
+		isEnveloped, ok := isResEnvelopedByIdStr[idStr]
+		if !ok {
+			// The resource is not found in the list of selected resources as reported by the status of the placement object.
+			//
+			// This is not considered as an error as the resource might be absent due to consistency reasons (i.e., it has
+			// just been de-selected); the status back-reporter will skip the resource for now.
+			klog.V(2).InfoS("Skip status back-reporting for the resource; the resource is not found in the list of selected resources in the placement object", "work", workRef, "resourceIdentifier", resIdentifier)
+			return
+		}
+		if isEnveloped {
+			// The resource is enveloped; skip back-reporting.
+			klog.V(2).InfoS("Skip status back-reporting for the resource; the resource is enveloped", "work", workRef, "resourceIdentifier", resIdentifier)
+			return
+		}
+
+		// Note that applied resources should always have a valid identifier set; for simplicity reasons
+		// here the back-reporter will no longer perform any validation.
+		gvr := schema.GroupVersionResource{
+			Group:    resIdentifier.Group,
+			Version:  resIdentifier.Version,
+			Resource: resIdentifier.Resource,
+		}
+		nsName := resIdentifier.Namespace
+		resName := resIdentifier.Name
+		unstructured, err := r.hubDynamicClient.Resource(gvr).Namespace(nsName).Get(ctx, resName, metav1.GetOptions{})
+		if err != nil {
+			wrappedErr := fmt.Errorf("failed to retrieve the target resource for status back-reporting: %w", err)
+			klog.ErrorS(err, "Failed to retrieve the target resource for status back-reporting", "work", workRef, "resourceIdentifier", resIdentifier)
+			errs[pieces] = wrappedErr
+			return
+		}
+
+		// Set the back-reported status to the target resource.
+		statusWrapper := make(map[string]interface{})
+		if err := json.Unmarshal(manifestCond.BackReportedStatus.ObservedStatus.Raw, &statusWrapper); err != nil {
+			wrappedErr := fmt.Errorf("failed to unmarshal back-reported status: %w", err)
+			klog.ErrorS(err, "Failed to unmarshal back-reported status", "work", workRef, "resourceIdentifier", resIdentifier)
+			errs[pieces] = wrappedErr
+			return
+		}
+
+		// Note that if the applied resource has a status sub-resource, it is usually safe for us to assume that
+		// the original resource should also have a status sub-resource of the same format.
+		unstructured.Object["status"] = statusWrapper["status"]
+		_, err = r.hubDynamicClient.Resource(gvr).Namespace(nsName).UpdateStatus(ctx, unstructured, metav1.UpdateOptions{})
+		if err != nil {
+			// TO-DO (chenyu1): check for cases where the API definition is inconsistent between the member cluster
+			// side and the hub cluster side, and single out the errors as user errors instead.
```

Comment:
good point,  it seems that we probably should start small? like only start with job/deployment i.e. something with stable API.